cloudify_kubernetes/workflows.py

```diff
@@ -125,3 +127,75 @@ def update_resource_definition(node_instance_id,
+
+
+def refresh_and_store_token(ctx,
+                            kubernetes_cluster_node_instance_id,
+                            deployment_capability_name,
+                            service_account_node_instance_id,
+                            secret_token_node_instance_id,
+                            store_token_and_kubeconfig_id):
+
+    cluster_node_instance = get_node_instance_from_node_instance_id_or_node_id(
+        kubernetes_cluster_node_instance_id)
+    execute_node_instance_operation(
+        cluster_node_instance, 'cloudify.interfaces.lifecycle.poststart')
+    execute_node_instance_operation(
+        cluster_node_instance, 'cloudify.interfaces.lifecycle.check_drift')
+
+    create_secrets_kubernetes_config(deployment_capability_name)
+
+    service_account_node_instance = \
+        get_node_instance_from_node_instance_id_or_node_id(
+            service_account_node_instance_id)
+    execute_node_instance_operation(service_account_node_instance,
+                                    'cloudify.interfaces.lifecycle.update')
+    execute_node_instance_operation(service_account_node_instance,
+                                    'cloudify.interfaces.lifecycle.poststart')
+
+    secret_token_node_instance = \
+        get_node_instance_from_node_instance_id_or_node_id(
+            secret_token_node_instance_id)
+    execute_node_instance_operation(secret_token_node_instance,
+                                    'cloudify.interfaces.lifecycle.delete')
+    execute_node_instance_operation(secret_token_node_instance,
+                                    'cloudify.interfaces.lifecycle.creat')
+
+    store_token_and_kubeconfig_node_instance = \
+        get_node_instance_from_node_instance_id_or_node_id(
+            store_token_and_kubeconfig_id)
+    execute_node_instance_operation(store_token_and_kubeconfig_node_instance,
+                                    'cloudify.interfaces.lifecycle.creat')
+
+
+def create_secrets_kubernetes_config(deployment_capability_name):
+    client = get_rest_client()
+
+    capabilities = client.deployments.capabilities. \
+        get(ctx.deployment.id).get('capabilities', {})
+    kubernetes_config = capabilities.get(deployment_capability_name, {}) \
+        .get('file_content', {})
+    ctx.logger.info('This is the capability: {}'.format(kubernetes_config))
+
+    try:
+        client.secrets.create('kubernetes_config', str(kubernetes_config))
+    except CloudifyClientError as err:
+        ctx.logger.error('{}'.format(str(err)))
+
+
+def get_node_instance_from_node_instance_id_or_node_id(
+        provided_node_instance_id):
+    try:
+        desired_node_instance = ctx.get_node_instance(
+            provided_node_instance_id)
+    except RuntimeError:
+        desired_node_instance = None
+        for node_instance in ctx.get_node_instance:
+            if node_instance.node_id == provided_node_instance_id:
+                desired_node_instance = node_instance
```

Comment:
since you found what you were looking for
```suggestion
                desired_node_instance = node_instance
                break
```